### PR TITLE
Dev/track oracle regex to kwiversys

### PR DESCRIPTION
**Pull request by collinsr**
gcc 4.8.x doesn't properly implement regex; switch back to kwiversys::RegularExpression.

### CODE REVIEW

**Comment by dstoup** [comment]
Jenkins test this please

File: track_oracle/file_formats/aoi_utils/aoi_utils.cxx [hunk 1]
```diff
@@ -78,24 +76,25 @@ parse_aoi_string( const string& s,
...
 
   // start picking off the numbers
-  while ( std::regex_search( a, b, what, float_pair_re ) )
-  {
-    double x = std::stod( what.str(1) );
-    double y = std::stor( what.str(2) );
-    points.push_back( vgl_point_2d<double>( x, y ));
-    a = what[2].second;
-  }
+#error "aoi_utils.cxx needs to be reworked vis-a-vis regex capabilities"
```
Comment (linus-sherrill) [inline track_oracle/file_formats/aoi_utils/aoi_utils.cxx]: Won't this cause a fatal build error?

File: track_oracle/file_formats/file_format_base.cxx [hunk 2]
```diff
@@ -32,37 +32,45 @@ glob_to_regexp_string( const string& glob )
...
   string::iterator i = baseglob.begin();
   bool prev_slash=false, in_sqr_brackets=false;
   //assemble the Regexp string
   string re = "^"; // match the start of the string
   while (i != baseglob.end())
   {
-    if (*i=='\\' && !prev_slash)
+    const char& c = *i;
+    if (c =='\\' && !prev_slash)
       prev_slash = true;
```
Comment (linus-sherrill) [inline track_oracle/file_formats/file_format_base.cxx]: Should this have brackets around a single line block?

Comment (collinsr) [reply track_oracle/file_formats/file_format_base.cxx]: Probably... I didn't update the style when I ported it over from vxl. Update coming shortly.

**Comment by dstoup** [comment]
Jenkins test this please

**Pull request merged**